sycl/include/sycl/ext/intel/experimental/fpga_kernel_attributes.hpp

```diff
@@ -0,0 +1,154 @@
+inline constexpr pipelined_key::value_t<
+    pipeline_directive_or_initiation_interval>
+    pipelined;
+
+} // namespace ext::intel::experimental
+
+namespace ext::oneapi::experimental {
+template <>
+struct is_property_key<intel::experimental::streaming_interface_key>
+    : std::true_type {};
+template <>
+struct is_property_key<intel::experimental::register_map_interface_key>
+    : std::true_type {};
+template <>
+struct is_property_key<intel::experimental::pipelined_key> : std::true_type {};
+
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+    intel::experimental::streaming_interface_key,
+    intel::experimental::fpga_kernel_attribute<T, PropertyListT>>
+    : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+    intel::experimental::register_map_interface_key,
+    intel::experimental::fpga_kernel_attribute<T, PropertyListT>>
+    : std::true_type {};
+template <typename T, typename PropertyListT>
+struct is_property_key_of<
+    intel::experimental::pipelined_key,
+    intel::experimental::fpga_kernel_attribute<T, PropertyListT>>
+    : std::true_type {};
+
+namespace detail {
+template <>
+struct PropertyToKind<intel::experimental::streaming_interface_key> {
+  static constexpr PropKind Kind = StreamingInterface;
+};
+template <>
+struct PropertyToKind<intel::experimental::register_map_interface_key> {
+  static constexpr PropKind Kind = RegisterMapInterface;
+};
+template <> struct PropertyToKind<intel::experimental::pipelined_key> {
+  static constexpr PropKind Kind = Pipelined;
+};
+
+template <>
+struct IsCompileTimeProperty<intel::experimental::streaming_interface_key>
+    : std::true_type {};
+template <>
+struct IsCompileTimeProperty<intel::experimental::register_map_interface_key>
+    : std::true_type {};
+template <>
+struct IsCompileTimeProperty<intel::experimental::pipelined_key>
+    : std::true_type {};
+
+template <intel::experimental::streaming_interface_options_enum Stall>
+struct PropertyMetaInfo<
+    intel::experimental::streaming_interface_key::value_t<Stall>> {
+  static constexpr const char *name = "sycl-streaming-interface";
+  static constexpr intel::experimental::streaming_interface_options_enum value =
+      Stall;
+};
+template <intel::experimental::register_map_interface_options_enum Wait>
+struct PropertyMetaInfo<
+    intel::experimental::register_map_interface_key::value_t<Wait>> {
+  static constexpr const char *name = "sycl-register-map-interface";
+  static constexpr intel::experimental::register_map_interface_options_enum
+      value = Wait;
+};
+template <int Value>
+struct PropertyMetaInfo<intel::experimental::pipelined_key::value_t<Value>> {
+  static constexpr const char *name = "sycl-pipelined";
+  static constexpr int value = Value;
+};
+
+} // namespace detail
+} // namespace ext::oneapi::experimental
+} // __SYCL_INLINE_VER_NAMESPACE(_V1)
+} // namespace sycl
```

Comment:
Is it possible to add some tests for this new API?

Reply:
added one.  Is that enough?